wallet/wallet.go

```diff
@@ -3730,25 +3731,27 @@ func CreateWatchingOnly(db walletdb.DB, pubPass []byte,
 	)
 }
 
-func create(db walletdb.DB, pubPass, privPass, seed []byte,
-	params *chaincfg.Params, birthday time.Time, isWatchingOnly bool,
+func create(db walletdb.DB, pubPass, privPass []byte,
+	rootKey *hdkeychain.ExtendedKey, params *chaincfg.Params,
+	birthday time.Time, isWatchingOnly bool,
 	cb func(walletdb.ReadWriteTx) error) error {
 
-	if !isWatchingOnly {
-		// If a seed was provided, ensure that it is of valid length. Otherwise,
-		// we generate a random seed for the wallet with the recommended seed
-		// length.
-		if seed == nil {
-			hdSeed, err := hdkeychain.GenerateSeed(
-				hdkeychain.RecommendedSeedLen)
-			if err != nil {
-				return err
-			}
-			seed = hdSeed
+	// If no root key was provided, we create one now from a random seed.
+	// But only if this is not a watching-only wallet where the accounts are
+	// created individually from their xpubs.
+	if !isWatchingOnly && rootKey == nil {
```

Comment:
We should check the `rootKey` is a private key if one is given.

Reply:
Yes, good idea. Done.